str/associatr/get_cis_numpy_files.py

```diff
@@ -0,0 +1,248 @@
+
+        gene_pheno_cov['sample_id'] = gene_pheno_cov['sample_id'].astype(float)
+
+        gene_pheno_cov = gene_pheno_cov.to_numpy()
+        with hl.hadoop_open(
+            output_path(
+                f'pheno_cov_numpy/{version}/{cell_type}/{chromosome}/{gene}_pheno_cov.npy'
+            ),
+            'wb',
+        ) as f:
+            np.save(f, gene_pheno_cov)
+
+
+@click.option('--input-h5ad-dir', help='GCS directory to the input AnnData objects')
+@click.option(
+    '--input-pseudobulk-dir', help='GCS directory to the input pseudobulk CSV files'
+)
+@click.option('--input-cov-dir', help='GCS directory to the input covariate CSV files')
+@click.option('--chromosomes', help=' eg chr22, comma separated if multiple')
+@click.option('--cell-types', help='cell type, comma separated if multiple')
+@click.option('--cis-window', help='cis window size in bp')
+@click.option('--version', default='v1', help='version of the output files')
+@click.option('--job-cpu', default=2, help='Number of CPUs to use for the job')
+@click.option('--job-memory', default='standard', help='Memory to use for the job')
+@click.option('--job-storage', default='5G', help='Storage to use for the job')
+@click.option(
+    '--max-parallel-jobs',
+    type=int,
+    default=500,
+    help=('To avoid exceeding Google Cloud quotas, set this concurrency as a limit.'),
+)
+@click.option(
+    '--min-pct',
+    default=1,
+    help='filter out genes that are expressed in fewer than XX% of cells',
+)
+@click.option(
+    '--remove-samples-file',
+    default=None,
+    help='GCS path to the file containing the list of samples to remove',
+)
+@click.command()
+def main(
+    input_h5ad_dir,
+    input_pseudobulk_dir,
+    input_cov_dir,
+    chromosomes,
+    cell_types,
+    cis_window,
+    version,
+    job_cpu,
+    job_memory,
+    job_storage,
+    max_parallel_jobs,
+    min_pct,
+    remove_samples_file,
+):
+    """
+    Run cis window extraction and phenotype/covariate numpy object creation
+    """
+    b = get_batch(name='get cis_numpy files')
+
+    # Setup MAX concurrency by genes
+    _dependent_jobs: list[hb.batch.job.Job] = []
+
+    def manage_concurrency_for_job(job: hb.batch.job.Job):
+        """
+        To avoid having too many jobs running at once, we have to limit concurrency.
+        """
+        if len(_dependent_jobs) >= max_parallel_jobs:
+            job.depends_on(_dependent_jobs[-max_parallel_jobs])
+        _dependent_jobs.append(job)
+
+    for cell_type in cell_types.split(','):
+        for chrom in chromosomes.split(','):
+            init_batch()
+            chrom_len = hl.get_reference('GRCh38').lengths[chrom]
+            j = b.new_python_job(
+                name=f'Extract cis window & phenotype and covariate numpy object for {cell_type}: {chrom}'
+            )
+            j.image(
+                'australia-southeast1-docker.pkg.dev/cpg-common/images/scanpy:1.9.3'
+            )
```

Comment:
```suggestion
            j.image(image_path('australia-southeast1-docker.pkg.dev/cpg-common/images/scanpy:1.9.3'))
```

changing this to pull from config will remove hard coding if you later bump the version in the image